### ISSUE TITLE
Invalid vertex pairs for surface edges

### ISSUE THREAD

**Issue by mldichter** [NREL/EnergyPlus#7372]
Issue overview
--------------
I have encountered "not fully enclosed" warnings before for a zone surface with a very long edge, but this warning is not pairing valid vertices. The first vertex pair is two diagonally opposite vertices, and the Z coordinate of the second vertex is wrong. The other edge is made from joining a vertex with itself. The surface is properly defined, so something is clearly wrong. I put a link to CmplxGlz_SingleZone_Vacuum.idf example file with modified geometry.

```
   ** Warning ** CalculateZoneVolume: The Zone="ROOM 102" is not fully enclosed. To be fully enclosed, each edge of a surface must also be an edge on one other surface.
   **   ~~~   **   The zone volume was calculated using the floor area times ceiling height method where the floor and ceiling are the same except for the z-coordinates.
   **   ~~~   **   The surface    "ROOM102 SOUTH WALL" has an edge that is either not an edge on another surface or is an edge on three or more surfaces: 
   **   ~~~   **     Vertex start { 0.0000, 0.0000, 0.0000}
   **   ~~~   **     Vertex end   { 3.6739E-014, -300.0000, 1.0000E-002}
   **   ~~~   **   The surface    "ROOM102 SOUTH WALL" has an edge that is either not an edge on another surface or is an edge on three or more surfaces: 
   **   ~~~   **     Vertex start { 0.0000, 0.0000, 0.0000}
   **   ~~~   **     Vertex end   { 0.0000, 0.0000, 0.0000}
TRUNCATED
```

```
BuildingSurface:Detailed,
    Room102 South Wall,      !- Name
    WALL,                    !- Surface Type
    Room Wall - South,       !- Construction Name
    Room 102,                !- Zone Name
    Outdoors,                !- Outside Boundary Condition
    ,                        !- Outside Boundary Condition Object
    SunExposed,              !- Sun Exposure
    WindExposed,             !- Wind Exposure
    0.5,                     !- View Factor to Ground
    4,                       !- Number of Vertices
    0,                       !- Vertex 1 X-coordinate {m}
    300,                     !- Vertex 1 Y-coordinate {m}
    0.01,                    !- Vertex 1 Z-coordinate {m}
    0,                       !- Vertex 2 X-coordinate {m}
    300,                     !- Vertex 2 Y-coordinate {m}
    0,                       !- Vertex 2 Z-coordinate {m}
    0,                       !- Vertex 3 X-coordinate {m}
    0,                       !- Vertex 3 Y-coordinate {m}
    0,                       !- Vertex 3 Z-coordinate {m}
    0,                       !- Vertex 4 X-coordinate {m}
    0,                       !- Vertex 4 Y-coordinate {m}
    0.01;                    !- Vertex 4 Z-coordinate {m}
```

### Details
Some additional details for this issue (if relevant):
 - Version of EnergyPlus 9.0

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Defect file added (https://drive.google.com/file/d/1LUbjQIDpn5CM5l-mkz2rSFO4jv-IGleP/view?usp=sharing)
 - [x] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

**Comment by dareumnam** [comment]
The geometry of "CmplxGlz_SingleZone_Vacuum.idf" is not fully enclosed because the two edges of the ceiling are not the edges of the East and the North wall. To be fully enclosed, each edge of a surface must also be an edge on one other surface.
As you said, the error file shows the vertices that seem wrong. But these vertices are the same geometry when the original geometry is rotated 180 degrees. If you select "Absolute" for Field: Coordinate System in GlobalGeometryRules instead of "Relative", you will see the same vertices to the original model geometry in error file.
On a separate note, it seems like there is another issue when we focused the second warning message:
```
   **   ~~~   **   The surface    "ROOM102 SOUTH WALL" has an edge that is either not an edge on another surface or is an edge on three or more surfaces: 
   **   ~~~   **     Vertex start { 0.0000, 0.0000, 0.0000}
   **   ~~~   **     Vertex end   { 0.0000, 0.0000, 0.0000}
```
It seems like E+ thinks two vertices as one point when the edge (the distance of the two vertices) is too short. I will take a look and create a new issue regarding this. Would it be okay if I mention the link for the idf file you mentioned (https://drive.google.com/file/d/1LUbjQIDpn5CM5l-mkz2rSFO4jv-IGleP/view?usp=sharing) while I am creating the new issue?

**Comment by mldichter** [comment]
Yes. Please use the link.
    -Mitchal
________________________________

**Comment by mjwitte** [comment]
@dareumnam When closing an issue, please add a comment explaining why.

**Comment by dareumnam** [comment]
@mjwitte Thank you for letting me know. I closed this issue because the point that the author of this issue questioned was addressed in the comment above, and a new issue (#7442) that related to this issue was submitted. I will make sure to leave a comment explaining why before closing an issue from next time.
